packages/bot-web-ui/src/components/dashboard/bot-builder/toolbox/toolbox.tsx

```diff
@@ -0,0 +1,205 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Field as FormField, Formik, Form } from 'formik';
+import { Input, Icon, Text } from '@deriv/components';
+import { localize } from '@deriv/translations';
+import { ToolboxItems } from './toolbox-items';
+import { connect } from '../../../../stores/connect';
+import RootStore from 'Stores/index.js';
+
+interface SearchBoxProps {
+    is_search_loading: boolean;
+    onSearch: () => void;
+    onSearchBlur: () => void;
+    onSearchClear: (param: (field: string, value: any, shouldValidate?: boolean | undefined) => void) => void;
+    onSearchKeyUp: (param: () => void) => void;
+}
+
+const SearchBox = ({ is_search_loading, onSearch, onSearchBlur, onSearchClear, onSearchKeyUp }: SearchBoxProps) => (
+    <div className='db-toolbox__search'>
+        <Formik initialValues={{ search: '' }} onSubmit={onSearch}>
+            {({ submitForm, values: { search }, setFieldValue }) => (
+                <Form>
+                    <FormField name='search'>
+                        {({ field }: any) => (
+                            <Input
+                                {...field}
+                                className='db-toolbox__search-field'
+                                type='text'
+                                name='search'
+                                placeholder={localize('Search')}
+                                onKeyUp={() => onSearchKeyUp(submitForm)}
+                                onFocus={submitForm}
+                                onBlur={onSearchBlur}
+                                leading_icon={
+                                    (search &&
+                                        (is_search_loading ? (
+                                            <div className='loader' />
+                                        ) : (
+                                            <Icon
+                                                icon='IcCloseCircle'
+                                                onClick={() => onSearchClear(setFieldValue)}
+                                                color='secondary'
+                                            />
+                                        ))) ||
+                                    (!search && <Icon icon='IcSearch' />)
+                                }
+                            />
+                        )}
+                    </FormField>
+                </Form>
+            )}
+        </Formik>
+    </div>
+);
+
+interface ToolboxProps {
+    hasSubCategory: () => boolean;
+    is_mobile: boolean;
+    is_search_loading: boolean;
+    is_toolbox_open: boolean;
+    onMount: (param?: any) => void;
+    onSearch: () => void;
+    onSearchBlur: () => void;
+    onSearchClear: () => void;
+    onSearchKeyUp: () => void;
+    onToolboxItemClick: () => void;
+    onToolboxItemExpand: (index: number) => void;
+    onUnmount: () => void;
+    sub_category_index: any[];
```

Comment:
@rupato-deriv could you please type it?

Reply:
let me check @maryiafrantsava-binary

Reply:
currently its unavoidable, I am leaving a note to fix it later 🙏